### PR TITLE
fix(router): support non-NgFactory promise in loadChildren typings

### PR DESCRIPTION
**Pull request by filipesilva**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/router/src/config.ts [hunk 1]
```diff
@@ -97,8 +97,8 @@ export type ResolveData = {
  * @see `Route#loadChildren`.
  * @publicApi
  */
-export type LoadChildrenCallback = () => Type<any>| NgModuleFactory<any>|
-    Promise<NgModuleFactory<any>>| Promise<Type<any>>| Observable<Type<any>>;
+export type LoadChildrenCallback = () => Type<any>| NgModuleFactory<any>| Observable<Type<any>>|
+    Promise<NgModuleFactory<any>| Type<any>| any>;
```
Comment (filipesilva) [inline packages/router/src/config.ts]: @alxhub is there a something better than `Promise<any>` for JIT NgModules here?

Comment (filipesilva) [reply packages/router/src/config.ts]: Examples of errors due to the missing typings can be seen on the failing CLI tests: https://circleci.com/gh/angular/angular-cli/51697 and https://circleci.com/gh/angular/angular-cli/51700.

File: packages/router/src/config.ts [hunk 2]
```diff
@@ -97,8 +97,8 @@ export type ResolveData = {
  * @see `Route#loadChildren`.
  * @publicApi
  */
-export type LoadChildrenCallback = () => Type<any>| NgModuleFactory<any>|
-    Promise<NgModuleFactory<any>>| Promise<Type<any>>| Observable<Type<any>>;
+export type LoadChildrenCallback = () => Type<any>| NgModuleFactory<any>| Observable<Type<any>>|
```
Comment (alxhub) [inline packages/router/src/config.ts]: The `|any` in `Promise<NgModuleFactory | Type | any>` makes the other two redundant. They're useful for documentation purposes, I suppose.

**Review by alxhub — APPROVED** [review]

File: packages/router/src/config.ts (hunk 2 above)
Comment (filipesilva) [reply packages/router/src/config.ts]: Yeah that's why I didn't want to remove them. If you think it should be different, I'll change.

**Review by kara — APPROVED** [review]

**Pull request closed without merging**